### PR TITLE
Add cortex-A55-based QEMU virtual board

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
This PR introduces cortex-A55-based QEMU virtual board (SMP and non-SMP version)

This PR depends on QEMU PR [1]  (currently, it seems to have been approved but not merged into master branch)

[1]: https://www.mail-archive.com/qemu-devel@nongnu.org/msg922724.html

CC: @dbaluta @iuliana-prodan @paulstelian97

### CODE REVIEW

**Comment by iuliana-prodan** [comment]
This looks good @LaurentiuM1234 
Can you please look on compliance checks - https://github.com/zephyrproject-rtos/zephyr/actions/runs/3648463007/jobs/6161989178

**Comment by iuliana-prodan** [comment]
@dleach02 can you, please, add @LaurentiuM1234 to Zephyr project so he can have rights to add reviewers & other stuff :)
Thanks you!

**Review by dbaluta — DISMISSED** [review]

**Comment by carlocaione** [comment]
The CI needs to be green.

**Comment by andrei-edward-popa** [comment]
You are missing a CONFIG_USERSPACE=y in boards/arm64/qemu_cortex_a55/qemu_cortex_a55_smp_defconfig because CONFIG_MAX_THREAD_BYTES depends on CONFIG_USERSPACE, probably that's why you have some failed builds.

**Comment by LaurentiuM1234** [comment]
> This looks good @LaurentiuM1234
> Can you please look on compliance checks - https://github.com/zephyrproject-rtos/zephyr/actions/runs/3648463007/jobs/6161989178

Should be fixed now.


> You are missing a CONFIG_USERSPACE=y in boards/arm64/qemu_cortex_a55/qemu_cortex_a55_smp_defconfig because CONFIG_MAX_THREAD_BYTES depends on CONFIG_USERSPACE, probably that's why you have some failed builds.

Thanks ! Will try adding said config. I was assuming all of the tests were failing because aforementioned patch is not yet merged so `cortex-a55` is not a valid `cpu` value for qemu.

File: dts/bindings/cpu/arm,cortex-a55.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright 2022 NXP
```
Comment (povergoing) [inline dts/bindings/cpu/arm,cortex-a55.yaml:1]: There is a race about this binding https://github.com/zephyrproject-rtos/zephyr/pull/52873/files#diff-c63bbaf10db596182beebe1b661bfc749ead826d42eaba2c6aab55c2d303e189

Comment (LaurentiuM1234) [reply dts/bindings/cpu/arm,cortex-a55.yaml:1]: Thanks, will remove the binding from this PR and add the PR introducing it as a dependency.

**Comment by carlocaione** [comment]
@LaurentiuM1234 do you plan to keep working on this?

File: boards/arm64/qemu_cortex_a55/qemu_cortex_a55_smp.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+name: QEMU Emulation for Cortex-A55 SMP
+type: qemu
+simulation: qemu
+arch: arm64
+toolchain:
+  - zephyr
+  - cross-compile
+ram: 128
+testing:
+  default: true
```
Comment (galak) [inline boards/arm64/qemu_cortex_a55/qemu_cortex_a55_smp.yaml:13]: Does this platform get us anything in CI that isn't convered by qemu_cortex_a53?

**Review by galak — CHANGES_REQUESTED** [review]
Want to see a justification for having this be a testing default platform.  What does this get us over a53 [no issue w/the code being supported upstream, just about adding yet another platform for testing to ci]

**Comment by LaurentiuM1234** [comment]
Sorry for taking such a long time to reach a conclusion for this PR.

> Want to see a justification for having this be a testing default platform. What does this get us over a53 [no issue w/the code being supported upstream, just about adding yet another platform for testing to ci]

Initially I was under the impression that before adding support for a new board you should also add support for a qemu virtual board based on the same CPU as your board hence this is why I made this PR. Since that's not the case, I suppose the only reason to keep this is as an extra test case to detect if something breaks the arm64 support.


> @LaurentiuM1234 do you plan to keep working on this?

Since it's not required for adding support for a new board I'm somewhat on the fence regarding this. If the extra test case is not required then I'm going to close this.

**Comment by LaurentiuM1234** [comment]
Closing this for now. Will reopen later on if need be.

**Pull request closed without merging**